gramps/gen/lib/json_utils.py

```diff
@@ -0,0 +1,219 @@
+
+import orjson
+
+# ------------------------------------------------------------------------
+#
+# Gramps modules
+#
+# ------------------------------------------------------------------------
+
+import gramps.gen.lib as lib
+
+
+NoneType = type(None)
+
+
+class DataDict(dict):
+    """
+    A wrapper around a data dict that also provides an
+    object interface.
+    """
+
+    def __init__(self, data=None):
+        """
+        Wrap a data dict (raw data) or object
+        with an attribute API. If data is an
+        object, we use it to get the attributes.
+        """
+        if isinstance(data, dict):
+            super().__init__(data)
+        else:
+            super().__init__()
+            if data:
+                # Data is actually a Gramps object
+                self.update(object_to_data(data))
+
+    def __str__(self):
+        if "_object" not in self:
+            self["_object"] = data_to_object(self)
+        return str(self["_object"])
+
+    def __getattr__(self, key):
+        if key.startswith("_"):
+            raise AttributeError(
+                "this method cannot be used to access hidden attributes"
+            )
+
+        if "_object" in self:
+            return getattr(self["_object"], key)
+        elif key in self:
+            value = self[key]
+            if isinstance(value, dict):
+                return DataDict(value)
+            elif isinstance(value, list):
+                return DataList(value)
+            else:
+                return value
+        else:
+            self["_object"] = data_to_object(self)
+            return getattr(self["_object"], key)
+
+
+class DataList(list):
+    """
+    A wrapper around a data list.
+    """
+
+    def __getitem__(self, position):
+        value = super().__getitem__(position)
+        if isinstance(value, dict):
+            return DataDict(value)
+        elif isinstance(value, list):
+            return DataList(value)
+        else:
+            return value
+
+
+def convert_state_to_object(obj_dict):
+    _class = obj_dict.pop("_class")
+    cls = lib.__dict__[_class]
+    obj = cls.__new__(cls)
+    obj.set_object_state(obj_dict)
+    return obj
+
+
+def convert_object_to_state(obj):
+    return obj.get_object_state()
+
+
+def string_to_data(string: str | bytes) -> DataDict:
+    """
+    Convert a JSON string into its data representation.
+    """
+    return DataDict(orjson.loads(string))
+
+
+def string_to_dict(string: str | bytes) -> dict:
+    """
+    Convert a JSON string into its dict representation.
+    """
+    return orjson.loads(string)
+
+
+def dict_to_string(dict: dict) -> str | bytes:
+    """
+    Convert a dict into its JSON string representation.
+    """
+    return orjson.dumps(dict)
+
+
+def object_to_dict(obj):
+    """
+    Convert any Gramps lib object into its DataDict representation.
+    """
+    if isinstance(obj, (float, int, str, NoneType)):
+        return obj
+
+    elif isinstance(obj, (list, tuple)):
```

Comment:
```suggestion
    if isinstance(obj, (list, tuple)):
```
Else after return.

Reply:
Performance is the same. "Explicit is better than implicit."